R/arguments.R

```diff
@@ -87,6 +87,10 @@ set_mode <- function(object, mode) {
       )
     )
   }
+  #only check if object is a model_spec
+  if(inherits(object, "model_spec")) {
+    check_spec_mode_val(class(object)[1], mode)
+  }
```

Comment:
```suggestion
  check_spec_mode_val(class(object)[1], mode)
```

I think you should be able to do this check unconditionally. `object` should always be a model-spec